blockchain/chain.go

```diff
@@ -137,17 +122,49 @@ func removeChildNode(children []*blockNode, node *blockNode) []*blockNode {
 	return children
 }
 
+// BestState houses information about the current best block and other info
+// related to the state of the main chain as it exists from the point of view of
+// the current best block.
+//
+// The BestSnapshot method can be used to obtain access to this information
+// in a concurrent safe manner and the data will not be changed out from under
+// the caller when chain state changes occur as the function name implies.
+// However, the returned snapshot must be treated as immutable since it is
```

Comment:
Fair enough. I agree that the additional overhead should be avoided if possible. 

We'll just have to be vigilant during future reviews of PR's involving `BestState` to ensure prohibited state mutation doesn't pass through unnoticed.